### PR TITLE
Removes unnecessary code

### PR DESCRIPTION
**Pull request by chvollm**
## IMPORTANT 

* [x] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [x] The changes are covered by unit tests which follow the Arrange-Act-Assert syntax and the naming conventions such as is used [in these tests](../tree/develop/Tests/FluentAssertions.Equivalency.Specs/MemberMatchingSpecs.cs#L51-L430).
* [x] If the PR adds a feature or fixes a bug, please update [the release notes](../tree/develop/docs/_pages/releases.md) with a functional description that explains what the change means to consumers of this library, which are published on the [website](https://fluentassertions.com/releases).
* [x] If the PR changes the public API the changes needs to be included by running [AcceptApiChanges.ps1](../tree/develop/AcceptApiChanges.ps1) or [AcceptApiChanges.sh](../tree/develop/AcceptApiChanges.sh).
* [x] If the PR affects [the documentation](../tree/develop/docs/_pages), please include your changes in this pull request so the documentation will appear on the [website](https://www.fluentassertions.com/introduction).

#1823

### CODE REVIEW

File: Src/FluentAssertions/Primitives/StringEqualityValidator.cs [hunk 1]
```diff
@@ -44,26 +44,14 @@ private string GetMismatchSegmentForStringsOfDifferentLengths()
...
-                {
-                    // Subject is shorter so we point at its last character.
-                    // We would like to point at next character as it is the real
-                    // index of first mismatch, but we need to point at character existing in
-                    // subject, so the next best thing is the last subject character.
-                    indexOfMismatch = Math.Max(0, Subject.Length - 1);
-                }
-                else
-                {
-                    // If subject is longer we are sure that subject starts with expected
```
Comment (dennisdoomen) [inline Src/FluentAssertions/Primitives/StringEqualityValidator.cs:63]: What makes you think this scenario is not necessary anymore?

Comment (chvollm) [reply Src/FluentAssertions/Primitives/StringEqualityValidator.cs:63]: It was red in my VS coverage. I then debugged all corresponding tests, e.g. `When_string_start_is_compared_with_string_that_is_longer_it_should_throw()` , they all don't go into the else branch. I even added a new test to be sure. 
Looking at `IndexOfFirstMismatch()`, it compares Subject with Expected by iterating over the characters of Subject. So the only scenario for -1 can be if Subject is shorter than Expected (or equal) and up until the end of Subject the characters are the same, then it stops at the end of Subject and doesn't compare the rest. Otherwise, if Subject is longer than Expected, we get the position of the first differing character.

Comment (jnyrup) [reply Src/FluentAssertions/Primitives/StringEqualityValidator.cs:63]: I haven't dug into the logic, but have you tried wrapping your tests in an assertion scope to if the logic handles continuing after the first failure?

The commit/PR adding the logic maybe also has some useful insights.

Comment (dennisdoomen) [reply Src/FluentAssertions/Primitives/StringEqualityValidator.cs:63]: It seems that @chvollm is right. That branch isn't covered by any tests and I can't seem to find an edge case where it fails.

**Review by dennisdoomen — APPROVED** [review]

File: Src/FluentAssertions/Primitives/StringEqualityValidator.cs (hunk 1 above)
Comment (chvollm) [reply Src/FluentAssertions/Primitives/StringEqualityValidator.cs:63]: Thanks. I spent quite some time on it but was still a bit unsure, so good to see that you can confirm my observation.

**Review by jnyrup — APPROVED** [review]
This code was added in #915.
https://github.com/fluentassertions/fluentassertions/pull/915/files#diff-ca701e8c572b2fa147076087dff48e19fc01b574423edabca1cc8c44e9af96a6R58-R75 adds three possible paths, but only two tests was added.

I tried running this, which should exercise the possible combinations of two strings not being equal.
None of the cases hit 
```cs
[Theory]
[InlineData(null, "")]
[InlineData(null, "a")]
[InlineData("", null)]
[InlineData("", "a")]
[InlineData("a", null)]
[InlineData("a", "")]
[InlineData("a", "aa")]
[InlineData("aa", "a")]
public void MyTestMethod(string a, string b)
{
    // Act
    var ass = new AssertionScope();
    a.Should().Be(b);
}
```

I'm confident with this change 👍
